rewrite-java/src/main/java/org/openrewrite/java/UseStaticImport.java

```diff
@@ -51,10 +51,15 @@ public String getDescription() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        int indexSpace = methodPattern.indexOf(' ');
-        int indexBrace = methodPattern.indexOf('(', indexSpace);
-        String identicalMethodName = "* " + methodPattern.substring(indexSpace, indexBrace) + "(..)";
-        return Preconditions.check(Preconditions.and(new UsesMethod<>(methodPattern), Preconditions.not(new DeclaresMethod<>(identicalMethodName))), new UseStaticImportVisitor());
+        TreeVisitor<?, ExecutionContext> preconditions = new UsesMethod<>(methodPattern);
+        if (!methodPattern.contains(" *(")) {
```

Comment:
I wasn't aware we did this kind of parsing in this recipe. Would be cleaner to use `MethodSignatureParser`, but probably a bit more work.

Reply:
Yes that was a recent addition; I had a brief look at using `MethodSignatureParser`, but as that takes a `TokenStream` I don't think it's easily adopted here. I don't think `" *("` could mismatch easily here; at worst we'd then not use a static import. It works for the downstream tests in rewrite-testing-frameworks.